azkaban-common/src/main/java/azkaban/utils/Props.java

```diff
@@ -811,6 +812,30 @@ public void storeFlattened(OutputStream out) throws IOException {
   }
 
   /**
+   * Returns a map of all the flattened properties
+   *
+   * @Return
+   */
+  public Map<String,String> getFlattened(){
```

Comment:
The new one returns a set that is sorted by the key value, which the old one doesn't , plus the old function seems need some reflecting work as the logic seems not trivial therefore I created this new one.  Second thought I think it is a better idea to fix the old one rather than creating a new one . I'll fix that .